ext/QuantumToolboxChainRulesCore.jl

```diff
@@ -0,0 +1,14 @@
+module QuantumToolboxChainRulesCore
+
+using LinearAlgebra
+import QuantumToolbox: QuantumObject
+using ChainRulesCore
```

Comment:
I would still prefer to list out the functions we used in `ChainRulesCore`

```suggestion
import ChainRulesCore: ...
```